### PR TITLE
Use Boost.Endian instead of custom wrappers

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

**Comment by HowardHinnant** [comment]
Missed a couple: https://github.com/HowardHinnant/rippled/commit/ee36be0376fbece39638fd1ea106b38a68916f44

File: src/ripple/basics/base_uint.h [hunk 1]
```diff
@@ -241,10 +241,15 @@ class base_uint
...
-    static uint32_t hostToBigend (uint32_t x) { return htobe32(x); }
+    static uint32_t bigendToHost (uint32_t x)
+    {
+        return boost::endian::big_to_native(x);
+    }
+
+    static uint32_t hostToBigend (uint32_t x)
+    {
+        return boost::endian::native_to_big(x);
+    }
```
Comment (HowardHinnant) [inline src/ripple/basics/base_uint.h]: These aren't used anywhere outside of `base_uint`.  How about removing these and just using `boost::endian` directly?

**Comment by scottschurr** [comment]
FWIW, this pull request is not currently building for me on macOS Sierra.  I'm getting this error message:
```
[ 34%] Building CXX object CMakeFiles/rippled.dir/src/ripple/overlay/impl/PeerImp.cpp.o
In file included from /Users/scottschurr/Development/rippled/src/ripple/overlay/impl/PeerImp.cpp:20:
In file included from /Users/scottschurr/Development/rippled/src/ripple/overlay/impl/PeerImp.h:27:
In file included from /Users/scottschurr/Development/rippled/src/ripple/overlay/impl/ProtocolMessage.h:23:
In file included from /Users/scottschurr/Development/rippled/build/cmake/clang.debug.nounity/ripple.pb.h:23:
In file included from /usr/local/include/google/protobuf/message.h:120:
/usr/local/include/google/protobuf/descriptor.h:358:5: error: expected
      identifier
    TYPE_BOOL           = 8,   // bool, varint on the wire.
    ^
/usr/include/ConditionalMacros.h:147:39: note: expanded from macro 'TYPE_BOOL'
     #define TYPE_BOOL                1
                                      ^
1 error generated.
make[2]: *** [CMakeFiles/rippled.dir/src/ripple/overlay/impl/PeerImp.cpp.o] Error 1
```
I'm not precisely sure what's going on but `PeerImp.h` used to #include `ByteOrder.h`, and no longer does.  And `ByteOrder.h` used to #include `ripple/beast/core/Config.h` which includes `ripple/beast/core/StandardConfig.h`.  `StandardConfig.h` does some horsing around with `TYPE_BOOL`.  Just a guess that might have something to do with it, not sure.

**Comment by HowardHinnant** [comment]
Thanks Scott!  I'm building on High Sierra but getting unittest failures.  I'm in the middle of tracking them down and you've given me new clues.

**Comment by nbougalis** [comment]
Ugh... all those files and their stupid interdependencies.

**Comment by HowardHinnant** [comment]
If everyone can test this fix, I'd appreciate it:  https://github.com/HowardHinnant/rippled/commit/bd3437fac8b8cea69a852b50a7a4ac69729fd166

This appears to fix a run time error in unittests for me, but I'm don't think it will fix the compile-time error Scott has reported.

Scott's error appears to be the lack of an `#undef TYPE_BOOL`.  @scottschurr  Is that a unity or non-unity build you're seeing that on?

**Comment by scottschurr** [comment]
@HowardHinnant, I cherry-picked your fix.  As expected I'm still running into the `TYPE_BOOL` problem.  That's on a non-unity build.  I also tried a unity build and got the same result, although the outermost reported file was `overlay2.cpp` for the unity build.  Macros continue to earn their reputation.

I'll horse around with adding includes from beast just as a way to characterize the problem.  If I can get it to build by adding an include that will tell us something.  We can tune from there.

**Comment by scottschurr** [comment]
Here are a couple of minimal changes that allow this branch to build and pass unit tests on macOS Sierra: https://github.com/scottschurr/rippled/commit/03baaf15378f78d3a3fec94b1615022928fdf134.  That's built on top of @HowardHinnant's fix.  I'm assuming that `ripple.pb.h` is generated, so we don't want to mess with its contents.  There may be a better approach...

**Comment by scottschurr** [comment]
Looks good as far as I can see.  Integer sizes look like they match expectations.  Builds and passes tests for me on macOS.

**Comment by nbougalis** [comment]
Merged as 3aaf6d785766cf37a6731ea28fa2b174ac1b28f5

**Comment by HowardHinnant** [comment]
Sorry, this dropped off my radar.  Thanks Scott for approving, and thanks Nik for merging!

**Pull request closed without merging**